app/routes/me.tsx

```diff
@@ -27,13 +30,38 @@ export default function UserPage() {
     }
   });
 
+  const [modalIsOpen, setModalIsOpen] = useState(false);
+
+  const openModal = () => {
+    setModalIsOpen(true)
+  }
+
+  const closeModal = () => {
+    setModalIsOpen(false)
+  }
+
   return (
     <div className="flex h-full min-h-screen flex-col">
       <header className="flex items-center justify-between bg-slate-800 p-4 text-white">
         <h1 className="text-3xl font-bold">
           Hello
         </h1>
         <p>{user.email}</p>
+        <div>
```

Comment:
can we break the modal out into its own component?   (does NOT need to be in its own file, just stick it at the bottom of the file for now)

Reply:
Just completed this.